bgpd/bgp_network.c

```diff
@@ -730,7 +731,8 @@ int bgp_socket(struct bgp *bgp, unsigned short port, const char *address)
 	if (bgpd_privs.change(ZPRIVS_LOWER))
 		flog_err(LIB_ERR_PRIVILEGES, "Can't lower privileges");
 	if (ret != 0) {
-		flog_err(LIB_ERR_SOCKET, "getaddrinfo: %s", gai_strerror(ret));
+		flog_err_sys(LIB_ERR_SOCKET, "getaddrinfo: %s",
+			     gai_strerror(ret));
```

Comment:
`errno` is not valid for `getaddrinfo()` failures, all of the error information is in the return value.

Reply:
(except if it returns `EAI_SYSTEM`, then `errno` is valid... should be the minority case though.)